src/chain/tezos/contracts/HicNFT.ts

```diff
@@ -0,0 +1,232 @@
+            }
+        ).join("; ");
+        return `{ ${transferList} }`
+    }
+
+    /*
+     * hDao transaction parameters
+     */
+    export type HDaoTransaction = TokenTransaction;
+
+    /*
+     * hDao transfer entrypoint parameters
+     */
+    export type HDaoTransferPair = TransferPair;
+
+    /*
+     * Helper function for single (i.e. non-batched) hDao FA2 transfer
+     *
+     * @param src
+     * @param dest
+     * @param amount
+     */
+    export function makeSingleHDaoTransfer(src: string, dest: string, amount: number): HDaoTransferPair {
+        return {
+            source: src,
+            txs: [ { destination: dest, token_id: 0, amount: amount } ]
+        }
+    }
+
+    /*
+     * Invokes the Transfer entrypoint of the hDao FA2 contract.
+     *
+     * @param server
+     * @param signer
+     * @param keystore
+     * @param address
+     * @param parameter
+     * @param amount
+     * @param fee
+     * @param gas
+     * @param freight
+     */
+    export async function hDaoTransfer(server: string, signer: Signer, keystore: KeyStore, transfers: HDaoTransferPair[], amount: number, fee: number, gas: number, freight: number): Promise<string> {
+        const entrypoint = `transfer`;
+
+        let parameter: string = TransferPairMichelson(transfers);
+        const nodeResult = await TezosNodeWriter.sendContractInvocationOperation(
+            server,
+            signer,
+            keystore,
+            hDaoAddress, // 'KT1AFA2mwNUMNd4SsujE1YYp29vd8BZejyKW'
+            amount,
+            fee,
+            freight,
+            gas,
+            entrypoint,
+            parameter,
+            TezosParameterFormat.Michelson);
+        return TezosContractUtils.clearRPCOperationGroupHash(nodeResult.operationGroupID);
+    }
+
+    /*
+     * Objkts transaction parameters
+     */
+    export type ObjktsTransaction = TokenTransaction;
+
+    /*
+     * Objkts transfer entrypoint parameters
+     */
+    export type ObjktsTransferPair = TransferPair;
+
+    /*
+     * Helper function for single (i.e. non-batched) objkts FA2 transfer
+     *
+     * @param src
+     * @param dest
+     * @param amount
+     */
+    export function makeSingleObjktsTransfer(src: string, dest: string, token_id: number, amount: number): ObjktsTransferPair {
+        return {
+            source: src,
+            txs: [ { destination: dest, token_id: token_id, amount: amount } ]
+        }
+    }
+
+    /*
+     * Invokes the Transfer entrypoint of the hDao FA2 contract.
+     *
+     * @param server
+     * @param signer
+     * @param keystore
+     * @param address
+     * @param parameter
+     * @param amount
+     * @param fee
+     * @param gas
+     * @param freight
+     */
+    export async function objktsTransfer(server: string, signer: Signer, keystore: KeyStore, transfers: HDaoTransferPair[], amount: number, fee: number, gas: number, freight: number): Promise<string> {
```

Comment:
@itamarreif: it's likely that [MultiAssetTokenHelper](https://github.com/Cryptonomic/ConseilJS/blob/master/src/chain/tezos/contracts/tzip12/MultiAssetTokenHelper.ts) can be used intead.